lib/timeline/Core.js

```diff
@@ -503,9 +497,9 @@ Core.prototype.setOptions = function (options) {
     this.configurator.setOptions(options.configure);
 
     // collect the settings of all components, and pass them to the configuration system
-    var appliedOptions = util.deepExtend({}, this.options);
+    var appliedOptions = { ...this.options };
     this.components.forEach(function (component) {
-      util.deepExtend(appliedOptions, component.options);
+      appliedOptions = { ...appliedOptions, ...component.options }
```

Comment:
I'm not sure if a destructuring can replace a `deepExtend`. This is discussed [here](https://github.com/visjs/vis-util/issues/14).

Reply:
It definitely can't. Unless the objects always have depth 1, that is `{ foo: true }` not `{ foo: { bar: false } }`.

Reply:
@yotamberk Please keep all the `deepExtend` and `selectiveDeepExtend` function. They can not simply be replaced by array destructuring.